packages/flutter_tools/lib/src/project.dart

```diff
@@ -472,8 +472,10 @@ class AndroidProject extends FlutterProjectPlatform {
       return false;
     }
     for (final String line in appGradle.readAsLinesSync()) {
-      if (line.contains(RegExp(r'apply from: .*/flutter.gradle')) ||
-          line.contains("def flutterPluginVersion = 'managed'")) {
+      final bool legacyApply = line.contains(RegExp(r'apply from: .*/flutter.gradle'));
```

Comment:
Instead of legacy can this be `fileBasedApply` then `apply` can be `declarativeApply` it give some more meaning to each of the evaluations.

Reply:
Done :)